src/sentry/tagstore/legacy/backend.py

```diff
@@ -110,6 +125,34 @@ def get_tag_values(self, project_ids, key, values=None):
 
         return list(qs)
 
+    def get_group_tag_key(self, group_id, key):
+        from sentry.tagstore.exceptions import GroupTagKeyNotFound
+
+        try:
+            return GroupTagKey.objects.get(
+                group_id=group_id,
+                key=key,
+            )
+        except GroupTagKey.DoesNotExist:
+            raise GroupTagKeyNotFound
+
+    def get_group_tag_keys(self, group_ids, keys=None, limit=None):
+        if isinstance(group_ids, list):
+            qs = GroupTagKey.objects.filter(group_id__in=group_ids)
+        else:
+            qs = GroupTagKey.objects.filter(group_id=group_ids)
+
+        if keys is not None:
+            if isinstance(keys, list):
```

Comment:
Sorta the same comment here. The problem is passing in a tuple or a set or somethign else, it'd yield unexpected behavior. Might want to compare against `six.text_types` instead of rigid check for `list`